### PR TITLE
[Modules] VM automanage update

### PR DESCRIPTION
**Pull request by eriqua**
# Description

Closes issue #965 

- Updated the Microsoft.Automanage/configurationProfileAssignments provider to be a resource within the vm module template instead of a separate module. That is not needed since no loop is required. One VM seems to allow only one configurationProfile. The array parameter configurationProfiles has thus been replaced by a configurationProfile string.
- Created dedicated parameter tests for linux and windows vms with automanage. Removing extensions deployment from those tests since having both applied created conflicts (e.g. the extension backup using a dependency rsv while the backup triggered by automanage using a dedicated rsv created together with the automanage setup deployment)
- Updated OS win versions an vm names
- Added module considerations for automanage

| Pipeline |
| - |
| [Compute: VirtualMachines](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.virtualmachines.yml) |

# Type of Change

Please delete options that are not relevant.

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: arm/Microsoft.Compute/virtualMachines/.parameters/linux.autmg.parameters.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{
```
Comment (MrMCake) [inline arm/Microsoft.Compute/virtualMachines/.parameters/linux.autmg.parameters.json:1]: Must be referenced in Azure DevOps pipelines

File: arm/Microsoft.Compute/virtualMachines/.parameters/windows.min.parameters.json [hunk 2]
```diff
@@ -2,11 +2,14 @@
     "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
     "contentVersion": "1.0.0.0",
     "parameters": {
+        "name": {
+            "value": "<<namePrefix>>-vm-win-02"
```
Comment (MrMCake) [inline arm/Microsoft.Compute/virtualMachines/.parameters/windows.min.parameters.json:6]: How about
```suggestion
            "value": "<<namePrefix>>-vm-winmin-001"
```

File: arm/Microsoft.Compute/virtualMachines/.parameters/windows.autmg.parameters.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "name": {
+            "value": "<<namePrefix>>-vm-win-03"
```
Comment (MrMCake) [inline arm/Microsoft.Compute/virtualMachines/.parameters/windows.autmg.parameters.json:6]:
```suggestion
            "value": "<<namePrefix>>-vm-winatmg-001"
```

File: arm/Microsoft.Compute/virtualMachines/.parameters/linux.autmg.parameters.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "name": {
+            "value": "<<namePrefix>>-vm-linux-autmg-01"
```
Comment (MrMCake) [inline arm/Microsoft.Compute/virtualMachines/.parameters/linux.autmg.parameters.json:6]:
```suggestion
            "value": "<<namePrefix>>-vm-linatmg-001"
```

File: arm/Microsoft.Compute/virtualMachines/.parameters/windows.parameters.json [hunk 5]
```diff
@@ -12,7 +12,7 @@
             "value": {
```
Comment (MrMCake) [inline arm/Microsoft.Compute/virtualMachines/.parameters/windows.parameters.json:12]: Cannot create a suggestion for code you didn't change - hence will do it here:

```bicep
"name": {
  "value": "<<namePrefix>>-vm-win-001"
}
```

File: arm/Microsoft.Compute/virtualMachines/.parameters/linux.parameters.json [hunk 6]
```diff
@@ -210,11 +210,6 @@
             "value": {
```
Comment (MrMCake) [inline arm/Microsoft.Compute/virtualMachines/.parameters/linux.parameters.json:210]: Cannot create a suggestion for code you didn't change - hence will do it here:

```bicep
"name": {
  "value": "<<namePrefix>>-vm-lin-001"
}
```

File: arm/Microsoft.Compute/virtualMachines/.parameters/windows.min.parameters.json (hunk 2 above)
Comment (eriqua) [reply arm/Microsoft.Compute/virtualMachines/.parameters/windows.min.parameters.json:6]: VM name has a limit of 15 chars for Windows

File: arm/Microsoft.Compute/virtualMachines/.parameters/windows.parameters.json (hunk 5 above)
Comment (eriqua) [reply arm/Microsoft.Compute/virtualMachines/.parameters/windows.parameters.json:12]: we changed to 01 after allowing namePrefix to 5 chars (thus carml) due to the limitation of max 15 chars for win vm names

File: arm/Microsoft.Compute/virtualMachines/.parameters/windows.autmg.parameters.json (hunk 3 above)
Comment (eriqua) [reply arm/Microsoft.Compute/virtualMachines/.parameters/windows.autmg.parameters.json:6]: same as above

File: arm/Microsoft.Compute/virtualMachines/.parameters/linux.autmg.parameters.json (hunk 4 above)
Comment (eriqua) [reply arm/Microsoft.Compute/virtualMachines/.parameters/linux.autmg.parameters.json:6]: Linux vms do not have the same limitation

File: arm/Microsoft.Compute/virtualMachines/.parameters/linux.autmg.parameters.json (hunk 1 above)
Comment (eriqua) [reply arm/Microsoft.Compute/virtualMachines/.parameters/linux.autmg.parameters.json:1]: Done

**Review by MrMCake — APPROVED** [review]

**Pull request merged**